code/game/machinery/ftl_shieldgen.dm

```diff
@@ -67,28 +75,31 @@
 
 /obj/machinery/ftl_shieldgen/process()
 	if(stat & (BROKEN|MAINT))
-		charging_power = 0
+		charging_power = TRUE
 		update_icon()
 		update_physical()
 		return
-	if(power_charge < power_charge_max)		// if there's power available, try to charge
-		var/load = charge_rate		// FUCK SEC
-		power_terminal.add_load(load)
-		if(power_terminal.surplus() * GLOB.CELLRATE <= 0)
-			power_charge += power_charge_max-power_charge
-		else
+	if(on)
+		if(power_charge < power_charge_max)		// if there's power available, try to charge
+			var/load = charge_rate // FUCK seconds
```

Comment:
I made this pr when i was retarded so continuing to work on this after 1 month made me woozy